pkgs/development/python-modules/pyobjc-core/default.nix

```diff
@@ -0,0 +1,133 @@
+    done
+  '';
+in
+
+buildPythonPackage rec {
+  pname = "pyobjc_core";
+  version = "10.3.1";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-sgSoDMwHD5qz+K9COjolpv14fiKFCNAMTDD4rFOLpyA=";
+  };
+  nativeBuildInputs = [
+    xcodebuild
+    cctools
+  ];
+  buildInputs = [
+    objc4
+    apple_libffi
+    Foundation
+    GameplayKit
+    MetalPerformanceShaders
+  ];
+  hardeningDisable = [ "strictoverflow" ]; # -fno-strict-overflow is not supported in clang on darwin
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=deprecated-declarations" ];
+  prePatch = ''
+    # TODO: Make patch for setup.py
+    # ignore the manual include flag for ffi, appears that it needs a very specific ffi from sdk (needs confirmation)
+    substituteInPlace setup.py --replace-fail '"-I/usr/include/ffi"' '#"-I/usr/include/ffi"'
+    # make os.path.exists that can spoil objc4 return True
+    substituteInPlace setup.py --replace-fail 'os.path.join(self.sdk_root, "usr/include/objc/runtime.h")' '"/"'
+
+    # Turn off clang’s Link Time Optimization, or else we can’t recognize (and link) Objective C .o’s:
+    sed -r 's/"-flto=[^"]+",//g' -i setup.py
+    # Fix some test code:
+    grep -RF '"sw_vers"' | cut -d: -f1 | while IFS= read -r file ; do
+      sed -r "s+"sw_vers"+"/usr/bin/sw_vers"+g" -i "$file"
+    done
+
+    # TODO: make a patch for tests
+    substituteInPlace \
+      PyObjCTest/test_nsdecimal.py \
+      --replace-fail "Cannot compare NSDecimal and decimal.Decimal" "Cannot compare NSDecimal and (\\\\w+.)?Decimal"
+    substituteInPlace \
+      PyObjCTest/test_bundleFunctions.py \
+      --replace-fail "os.path.expanduser(\"~\")" "\"/var/empty\""
+    substituteInPlace \
+      PyObjCTest/test_methodaccess.py \
+      --replace-fail "testClassThroughInstance2" \
+      "disable_testClassThroughInstance2"
+  '';
+  preBuild = commonPreBuildDarwinMinVersion;
+  # XXX: We’re turning tests off, because they’re mostly working (0.54% failures among 4,600 tests),
+  # and I don’t have any more time to investigate now (maybe in a Nixpkgs contribution in the future):
+  #
+  # YYY: There were broken tests that relied on home paths and printing format, prePatch phase fixes or disables those
+  #
+  # pyobjc-core> Ran 4600 tests in 273.830s
+  # pyobjc-core> FAILED (failures=3, errors=25, skipped=4, expected failures=3, unexpected successes=1)
+  # pyobjc-core> SUMMARY: {'count': 4600, 'fails': 3, 'errors': 25, 'xfails': 3, 'xpass': 0, 'skip': 4}
+  # pyobjc-core> error: some tests failed
+  #
+  # ZZZ: Sorted out the issues with tests and put them in prePatch
+  checkPhase = ''
+    python3 setup.py test
+  '';
+  doCheck = true;
+  meta = {
+    description = "The Python <-> Objective-C Bridge with bindings for macOS frameworks";
+    homepage = "https://pypi.org/project/pyobjc-core/";
+    platforms = lib.platforms.darwin;
+  };
+  # Help derived packages to use dependencies as they were in pyobjc-core
+  passthru = {
```

Comment:
Meta is always last and passhtru straight before it.

Reply:
did that